src/table/table-adapter.class.ts

```diff
@@ -116,23 +127,47 @@ export class TableDomAdapter implements TableAdapter {
 	 * @param column index of the column
 	 */
 	getCell(row: number, column: number): HTMLTableCellElement {
-		const baseRow = this.getRow(row);
-		let baseCells = Array.from(baseRow.cells);
+		const col = this.getColumn(column);
 
-		// rows can have fewer total cells than the actual table
-		// the model pretends all rows behave the same (with colspans > 1 being N cells long)
-		// this maps that view to the HTML view (colspans > 1 are one element, so the array is shorter)
-		let realIndex = 0;
-		// i is only used for iterating the "cells"
-		for (let i = 0; i < column;) {
-			// skip the next N "cells"
-			i += baseCells[realIndex].colSpan;
-			// don't bump realIndex if i now exceeds the column we're shooting for
-			if (i > column) { break; }
-			// finally, increment realIndex (to keep it generally in step with i)
-			realIndex++;
+		return this.findCellInColumn(col, row).cell;
+	}
+
+	/**
+	 * Returns a column from the table, using the `id` and `headers` attributes
+	 *
+	 * See here for more detail these attributes: https://www.w3.org/TR/WCAG20-TECHS/H43.html
+	 *
+	 * @param column the index of the column
+	 */
+	getColumn(column: number): HTMLTableCellElement[] {
+		const firstHeader = Array.from(this.tableElement.rows[0].cells);
+
+		const { cell: header, realIndex: realColumnIndex } = this.findCellInRow(firstHeader, column);
+
+		const linkedCells: HTMLTableCellElement[] = [];
+
+		for (let i = 1; i < this.tableElement.rows.length; i++) {
```

Comment:
Can we use `forEach` here?

Reply:
Probably 🤔

Reply:
So for whatever reason `.rows` is a `HTMLCollection` instead of a `NodeList` ... so we can't `forEach` it - https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection

Reply:
I was suspecting that that's the reason you use `for`